### PR TITLE
Filter spliced reads for IndelRealignment

### LINKED ISSUE #111: Stop IndelRealigning anything, instead run left-normalize indels
cf https://github.com/hammerlab/biokepi/issues/390

### PR DESCRIPTION
**Pull request by ihodes**
We split RNA into two BAMs after realignment and some post-processing;
one with all the reads that span a splice junction (i.e. those with
CIGAR strings with an N in them). We IndelRealign unspliced reads, and
then merge the spliced reads back into the final BAM. Do we this because
the IndelRealigner will drop all reads in a pileup after a spliced read
is found, which can end up dropping the vast majority of our reads. We
want to continue running IndelRealignment, though, in order to retain as
much RNA support for indels as possible.

resolves https://github.com/hammerlab/epidisco/issues/111

### CODE REVIEW

File: src/lib/pipeline.ml [hunk 1]
```diff
@@ -251,11 +252,23 @@ module Full (Bfx: Extended_edsl.Semantics) = struct
...
+    let spliced_bam =
+      let filter = Biokepi.Tools.Sambamba.Filter.Defaults.only_split_reads in
+      Bfx.sambamba_filter ~filter bam in
+    let indel_realigned_bam =
+      let filter = Biokepi.Tools.Sambamba.Filter.Defaults.drop_split_reads in
+      Bfx.sambamba_filter ~filter bam
+      |> Bfx.gatk_indel_realigner
+        ~configuration:indel_realigner_config
+    in
+    Bfx.merge_bams @@ Bfx.list [spliced_bam; indel_realigned_bam]
```
Comment (iskandr) [inline src/lib/pipeline.ml:271]: Does `merge_bams` produce a sorted BAM?

Comment (ihodes) [reply src/lib/pipeline.ml:271]: Yes indeed.

**Review by smondet — APPROVED** [review]
The code LGTM, but how does this work around the probem of indels "aligned to the left"?
The non-split reads can have the same indels expressed differently so in the merged bam won't be consistent, right?

**Comment by ihodes** [comment]
Definitely the downside of this approach; the spliced reads at indels may very well be aligned differently than the indel-realigned non-spliced reads. This doesn't affect our pipeline, but is certainly worth noting.

**Pull request merged**